### PR TITLE
[ETHEREUM-CONTRACTS] Disable flownfts

### LINKED ISSUE #1993: [ETHEREUM-CONTRACTS] Disable FlowNFT hooks
## What & Why

FlowNFTs have not turned out to be as useful as hoped for, adding cosiderable complexity and gas cost overhead.

We thus decided to deprecate them. As a first step, the hooks in CFA and GDA are removed.

## How

* Remove the code for invoking hooks
* Adjust tests

### LINKED ISSUE #1995: [ETHEREUM-CONTRACTS] [METADATA] [SDK-CORE] remove flowNFT code
## What

Remove code and related tests from packages in protocol-monorepo.
This will touch several packages.

Note that SuperToken.sol currently has protection against flownft address changes in updateCode: https://github.com/superfluid-finance/protocol-monorepo/blob/dev/packages/ethereum-contracts/contracts/superfluid/SuperToken.sol#L190

This means that for the next protocol upgrade we need to leave enough code for the previous flownft proxy addresses to be provided as constructor argument.

## AC

- [ ] Remove all related code not needed for protocol upgrade
- [ ] Create new package releases affected by the change

### PR DESCRIPTION
**Pull request by d10r**
See #1993

deploy-framework.js won't deploy FlowNFT contracts anymore. But it will preserve existing contract addresses if set.

Note that only enough code to keep the CI pipeline green was removed, there's still a lot of FlowNFT related code left.
As is in this PR, it would be relatively easy to "bring it back" later.
In order to fully purge, a lot more code locations need to be touched.

piggy-backed:
* include previous canonical superTokenLogic in list of logic addresses to be updated from

### CODE REVIEW

**Review by hellwolf — COMMENTED** [review]
- remove eNFT completely from code base
- Remove all eNFT references in ISuperToken
- Make sure deployment scripts no longer dependent on the eNFT interfaces.

**Comment by hellwolf** [comment]
to be released as v1.11.0

File: packages/subgraph/src/mappings/flowNFT.ts [hunk 1]
```diff
@@ -1,67 +0,0 @@
...
-    ev.value = event.params.tokenId;
-    ev.token = event.address;
-
-    ev.save();
-
-    getOrInitAccount(event.params.to, event.block);
-    getOrInitAccount(event.params.from, event.block);
-}
-
-export function handleMetadataUpdate(event: MetadataUpdate): void {
```
Comment (kasparkallas) [inline packages/subgraph/src/mappings/flowNFT.ts:60]: Only the FlowNFT handling of the event disappears, right, not `MetadataUpdate` altogether?

File: packages/subgraph/schema.graphql [hunk 2]
```diff
@@ -1514,63 +1505,20 @@ type ApprovalEvent implements Event @entity(immutable: true) {
...
     """
-    The address that will be granted allowance to transfer ERC20/NFT.
+    The address that will be granted allowance to transfer ERC20.
     """
     to: Account!
 
-    """
-    Indicates whether the event was emitted for the approval of an NFT.
-    """
-    isNFTApproval: Boolean!
```
Comment (kasparkallas) [inline packages/subgraph/schema.graphql:1529]: We could arguably leave the NFT fields to reduce the chance of API usage breakage. In other words, right now the change is potentially a breaking change, but we could leave it as non-breaking. Just mark them obsolete and have them be falsy/default values.

File: packages/sdk-core/src/subgraph/queries/getAllEvents.graphql [hunk 3]
```diff
@@ -397,13 +397,6 @@ query getAllEvents(
         ... on MemberUnitsUpdatedEvent {
             ...eventFields
         }
-        ... on ApprovalEvent {
```
Comment (kasparkallas) [inline packages/sdk-core/src/subgraph/queries/getAllEvents.graphql:400]: This most likely shouldn't disappear here because of the ERC-20 Approval.

**Comment by d10r** [comment]
removal of an event / attributes (?) will break `getEvents()` in sdk-core.
Thus this order of events is important:
* create new sdk-core package and update products
* switch canonical subgraph to new schema

File: packages/subgraph/schema.graphql (hunk 2 above)
Comment (d10r) [reply packages/subgraph/schema.graphql:1529]: we discussed this and decided to do a breaking change instead of leaving tech debt

**Comment by d10r** [comment]
reason for `docker-compose` -> `docker compose`: https://github.com/actions/runner-images/issues/9692

File: packages/subgraph/src/mappings/flowNFT.ts (hunk 1 above)
Comment (d10r) [reply packages/subgraph/src/mappings/flowNFT.ts:60]: didn't realize this is FlowNFT specific too. So, yes, I guess more pruning...

Comment (d10r) [reply packages/subgraph/src/mappings/flowNFT.ts:60]: removed now

**Comment by hellwolf** [comment]
```
$ git grep -i flownft packages/ | wc -l
97

$ git grep -i flownft packages/ | grep -Fv CHANGELOG.md | head -n10
packages/ethereum-contracts/contracts/mocks/SuperTokenFactoryMock.t.sol:import { SuperTokenFactoryBase, IConstantOutflowNFT, IConstantInflowNFT } from "../superfluid/SuperTokenFactory.sol";
packages/ethereum-contracts/contracts/mocks/SuperTokenFactoryMock.t.sol:        IConstantOutflowNFT constantOutflowNFT,
packages/ethereum-contracts/contracts/mocks/SuperTokenFactoryMock.t.sol:        IConstantInflowNFT constantInflowNFT,
packages/ethereum-contracts/contracts/mocks/SuperTokenFactoryMock.t.sol:        SuperTokenFactoryBase(host, superTokenLogic, constantOutflowNFT, constantInflowNFT, poolAdminNFT, poolMemberNFT)
packages/ethereum-contracts/contracts/mocks/SuperTokenFactoryMock.t.sol:        IConstantOutflowNFT constantOutflowNFT,
packages/ethereum-contracts/contracts/mocks/SuperTokenFactoryMock.t.sol:        IConstantInflowNFT constantInflowNFT,
packages/ethereum-contracts/contracts/mocks/SuperTokenFactoryMock.t.sol:        SuperTokenFactoryBase(host, superTokenLogic, constantOutflowNFT, constantInflowNFT, poolAdminNFT, poolMemberNFT)
packages/ethereum-contracts/contracts/mocks/SuperTokenFactoryMock.t.sol:        IConstantOutflowNFT constantOutflowNFT,
packages/ethereum-contracts/contracts/mocks/SuperTokenFactoryMock.t.sol:        IConstantInflowNFT constantInflowNFT,
packages/ethereum-contracts/contracts/mocks/SuperTokenFactoryMock.t.sol:        SuperTokenFactoryBase(host, superTokenLogic, constantOutflowNFT, constantInflowNFT, poolAdminNFT, poolMemberNFT)

$...
```


What are these remnants?

**Comment by d10r** [comment]
The currently deployed `Supertoken.sol` has this in `updateCode()`:
```
        // @note This is another check to ensure that when updating to a new SuperToken logic contract
        // that we have passed the correct NFT proxy contracts in the construction of the new SuperToken
        // logic contract
        if (
            CONSTANT_OUTFLOW_NFT !=
            SuperToken(newAddress).CONSTANT_OUTFLOW_NFT() ||
            CONSTANT_INFLOW_NFT !=
            SuperToken(newAddress).CONSTANT_INFLOW_NFT()
        ) {
            revert SUPER_TOKEN_NFT_PROXY_ADDRESS_CHANGED();
        }
```

So we can't set the address to zero in this upgrade, meaning we can't fully prune in 1 iteration.
This PR removes this check, so in a next iteration we could complete the pruning.

Also I already removed `CONSTANT_OUTFLOW_NFT()` and `CONSTANT_OUTFLOW_NFT()` from `ISuperToken`.

**Review by hellwolf — APPROVED** [review]
:) great!

**Pull request merged**